E0070 needs to be updated to new format

From: src/test/compile-fail/E0070.rs

Error E0070 needs a span_label, updating it from:

```
error[E0070]: invalid left-hand side expression
  --> src/test/compile-fail/E0070.rs:18:5
   |
18 |     some_other_func() = 4; //~ ERROR E0070
   |     ^^^^^^^^^^^^^^^^^^^^^
```

To:

```
error[E0070]: invalid left-hand side expression
  --> src/test/compile-fail/E0070.rs:18:5
   |
18 |     some_other_func() = 4; //~ ERROR E0070
   |     ^^^^^^^^^^^^^^^^^^^^^ left-hand of expression not valid
```

--- a/src/librustc_typeck/check/mod.rs
+++ b/src/librustc_typeck/check/mod.rs
@@ -3463,8 +3463,13 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
 
             let tcx = self.tcx;
             if !tcx.expr_is_lval(&lhs) {
-                span_err!(tcx.sess, expr.span, E0070,
-                    "invalid left-hand side expression");
+                struct_span_err!(
+                    tcx.sess, expr.span, E0070,
+                    "invalid left-hand side expression")
+                .span_label(
+                    expr.span,
+                    &format!("left-hand of expression not valid"))
+                .emit();
             }
 
             let lhs_ty = self.expr_ty(&lhs);

--- a/src/test/compile-fail/issue-26093.rs
+++ b/src/test/compile-fail/issue-26093.rs
@@ -12,6 +12,7 @@ macro_rules! not_an_lvalue {
     ($thing:expr) => {
         $thing = 42;
         //~^ ERROR invalid left-hand side expression
+        //~^^ NOTE left-hand of expression not valid
     }
 }
 

